packages/flutter/test/material/list_tile_theme_test.dart

```diff
@@ -0,0 +1,445 @@
+  const TestIcon({ Key? key }) : super(key: key);
+
+  @override
+  TestIconState createState() => TestIconState();
+}
+
+class TestIconState extends State<TestIcon> {
+  late IconThemeData iconTheme;
+
+  @override
+  Widget build(BuildContext context) {
+    iconTheme = IconTheme.of(context);
+    return const Icon(Icons.add);
+  }
+}
+
+class TestText extends StatefulWidget {
+  const TestText(this.text, { Key? key }) : super(key: key);
+
+  final String text;
+
+  @override
+  TestTextState createState() => TestTextState();
+}
+
+class TestTextState extends State<TestText> {
+  late TextStyle textStyle;
+
+  @override
+  Widget build(BuildContext context) {
+    textStyle = DefaultTextStyle.of(context).style;
+    return Text(widget.text);
+  }
+}
+
+void main() {
+  test('ListTileThemeData copyWith, ==, hashCode basics', () {
+    expect(const ListTileThemeData(), const ListTileThemeData().copyWith());
+    expect(const ListTileThemeData().hashCode, const ListTileThemeData().copyWith().hashCode);
+  });
+
+  test('ListTileThemeData defaults', () {
+    const ListTileThemeData themeData = ListTileThemeData();
+    expect(themeData.dense, null);
+    expect(themeData.shape, null);
+    expect(themeData.style, null);
+    expect(themeData.selectedColor, null);
+    expect(themeData.iconColor, null);
+    expect(themeData.textColor, null);
+    expect(themeData.contentPadding, null);
+    expect(themeData.tileColor, null);
+    expect(themeData.selectedTileColor, null);
+    expect(themeData.horizontalTitleGap, null);
+    expect(themeData.minVerticalPadding, null);
+    expect(themeData.minLeadingWidth, null);
+    expect(themeData.enableFeedback, null);
+    expect(themeData.mouseCursor, null);
+    expect(themeData.visualDensity, null);
+  });
+
+  testWidgets('Default ListTileThemeData debugFillProperties', (WidgetTester tester) async {
+    final DiagnosticPropertiesBuilder builder = DiagnosticPropertiesBuilder();
+    const ListTileThemeData().debugFillProperties(builder);
+
+    final List<String> description = builder.properties
+      .where((DiagnosticsNode node) => !node.isFiltered(DiagnosticLevel.info))
+      .map((DiagnosticsNode node) => node.toString())
+      .toList();
+
+    expect(description, <String>[]);
+  });
+
+  testWidgets('ListTileThemeData implements debugFillProperties', (WidgetTester tester) async {
+    final DiagnosticPropertiesBuilder builder = DiagnosticPropertiesBuilder();
+    const ListTileThemeData(
+      dense: true,
+      shape: StadiumBorder(),
+      style: ListTileStyle.drawer,
+      selectedColor: Color(0x00000001),
+      iconColor: Color(0x00000002),
+      textColor: Color(0x00000003),
+      contentPadding: EdgeInsets.all(100),
+      tileColor: Color(0x00000004),
+      selectedTileColor: Color(0x00000005),
+      horizontalTitleGap: 200,
+      minVerticalPadding: 300,
+      minLeadingWidth: 400,
+      enableFeedback: true,
+      mouseCursor: MaterialStateMouseCursor.clickable,
```

Comment:
visualDensity?

Reply:
I will need to change this test to this style to test `visualDensity` since it provides a hash code. 

https://github.com/flutter/flutter/blob/3420b9c50ea19489dd74b024705bb010c5763d0a/packages/flutter/test/material/radio_theme_test.dart#L69

Existing test doesn't seem compatible with `equalsIgnoringHashCodes` function.